### ISSUE TITLE
Possible escaping issues

### ISSUE THREAD

**Issue by martindholmes** [UVicHCMC/jsontoxml#3]
When you're writing out XML as text using a print approach, there are all sorts of potential issues that may cause problems, and using a serializer such as dicttoxml is usually a safer solution. But if you're writing the serializer yourself as you are here, these are some things to watch out for:

 - open angle bracket must be an entity
 - close angle bracket probably should be, although not mandatory, 
 - the ampersand must be an entity when it is a literal ampersand; but beware of its use as part of another existing entity, so if you have (for example) `&#x2014`, then you obviously don't want to escape the ampersand, but if you have `Marks & Spencer`, then you do
 -  the single apostrophe and double quote should probably be entities (especially in attribute values that might be delimited with them)
 - embedded markup such as HTML fragments.
 - any Unicode character from the surrogate blocks, U+FFFE, and U+FFFF.
 
 Also character encoding is always a concern; presumably you want to output UTF-8, but if your data is coming from a db then the source may not be in UTF-8, so make sure that accented characters etc. are encoded correctly.

**Comment by daydin** [comment]
Now that I'm using ElementTree to add nodes and parse out the XML, I'm not sure if this is still relevant. Parts of it immediately seem to be, like the UTF-8 encoding consideration, and handling of special characters and apostrophes. But I am not sure if any of the first three bullet points, as well as the final bullet point, still apply.

**Comment by martindholmes** [comment]
@daydin Yes, the serializers built into ElementTree should handle all of this for you. Closing this!
